libs/feature/catalog/src/lib/source-label/source-label.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, ChangeDetectionStrategy, Input } from '@angular/core'
+import { LANG_2_TO_3_MAPPER } from '@geonetwork-ui/util/i18n'
+import { TranslateService } from '@ngx-translate/core'
+import { Observable } from 'rxjs'
+import { map } from 'rxjs/operators'
+import { CatalogSource } from '../sources/sources.model'
+import { SourcesService } from '../sources/sources.service'
+
+@Component({
+  selector: 'gn-ui-source-label',
+  templateUrl: './source-label.component.html',
+  styleUrls: ['./source-label.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SourceLabelComponent {
+  sourceLabel$: Observable<CatalogSource>
+  @Input() set catalogUuid(uuid: string) {
+    this.sourceLabel$ = this.sourcesService
+      .getSource(uuid)
+      .pipe(
+        map(
+          (source) =>
+            source.label[LANG_2_TO_3_MAPPER[this.translateService.currentLang]]
```

Comment:
Can you cherry-pick https://github.com/geonetwork/geonetwork-ui/pull/218/commits/b731b0aebb3f29fe5dd717bb281fc5845f040860 from multilingual PR and use this service instead plz, thanks